### PR TITLE
reqs: Improve pip command

### PR DESCRIPTION
**Pull request by kingiler**
It is recommanded to use `python -m pip` than plain `pip`. As in some cases `pip` may not come from the virtualenv while the former ensure the `pip` is bundled with the `python` used.

### CODE REVIEW

**Comment by AntonLydike** [comment]
Why is is preferable to use the pip *not* from the virtualenv? Shouldn't you want to use that one?

**Comment by kingiler** [comment]
What I mean is the `pip` may not come from the virtualenv, which I encounter a couple of times.

**Comment by superlopuh** [comment]
I would recommend using `--require-venv` instead

**Comment by AntonLydike** [comment]
Oooh, sorry, misunderstood that PR. Looks good, `--require-venv` may even be better!

**Comment by kingiler** [comment]
Tested on my computer, `--require-virtualenv` seems not getting what is intended.

**Review by AntonLydike — APPROVED** [review]

**Comment by kingiler** [comment]
`--require-virtualenv` is added for completeness.

**Review by georgebisbas — DISMISSED** [review]
Why is this change happening?
Our ci setup looks fine with this.
Is this a problem you encountered locally?

**Comment by georgebisbas** [comment]
Also, kindly check this: https://github.com/xdslproject/xdsl/wiki/Tags-for-commit-messages-and-PR-titles
to prepare your PR

**Review by math-fehr — APPROVED** [review]
:+1:

**Comment by kingiler** [comment]
@georgebisbas, yes a little problem using the `make venv`.

**Comment by georgebisbas** [comment]
> @georgebisbas, yes a little problem using the `make venv`.
What was the problem? Can you paste the log?

File: Makefile [hunk 1]
```diff
@@ -88,5 +88,5 @@ coverage-report-md:
 venv: requirements-optional.txt requirements.txt
 	python3 -m venv ${VENV_DIR}
 	source ${VENV_DIR}/bin/activate
-	pip install -r requirements-optional.txt -r requirements.txt
-	pip install -e ".[extras]"
+	python3 -m pip --require-virtualenv install -r requirements-optional.txt -r requirements.txt
```
Comment (superlopuh) [inline Makefile:91]: why the python3?

Comment (kingiler) [reply Makefile:91]: Just be consistent with line 89 `python3 -m venv ${VENV_DIR}`

Comment (superlopuh) [reply Makefile:91]: ah didn't spot that... @AntonLydike is that so people don't run it with Python 2 or Python 4? :)

Comment (AntonLydike) [reply Makefile:91]: Just a weird habit I picked up a while back. Maybe related to some work on ancient servers with questionable defaults^^

**Review by superlopuh — APPROVED** [review]

**Review by georgebisbas — APPROVED** [review]

**Pull request merged**